x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/cel_input_step/generation_modal.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import React, { useCallback, useEffect, useState } from 'react';
+import { css } from '@emotion/react';
+import { getLangSmithOptions } from '../../../../../common/lib/lang_smith';
+import { type CelInputRequestBody } from '../../../../../../common';
+import { runCelGraph } from '../../../../../common/lib/api';
+import { useKibana } from '../../../../../common/hooks/use_kibana';
+import type { State } from '../../state';
+import * as i18n from './translations';
+import { useTelemetry } from '../../../telemetry';
+
+export type OnComplete = (result: State['celInputResult']) => void;
+
+interface UseGenerationProps {
+  integrationSettings: State['integrationSettings'];
+  connector: State['connector'];
+  onComplete: OnComplete;
+}
+export const useGeneration = ({
+  integrationSettings,
+  connector,
+  onComplete,
+}: UseGenerationProps) => {
+  const { reportCelGenerationComplete } = useTelemetry();
+  const { http, notifications } = useKibana().services;
+  const [error, setError] = useState<null | string>(null);
+  const [isRequesting, setIsRequesting] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (
+      !isRequesting ||
+      http == null ||
+      connector == null ||
+      integrationSettings == null ||
+      notifications?.toasts == null
+    ) {
+      return;
+    }
+    const generationStartedAt = Date.now();
+    const abortController = new AbortController();
+    const deps = { http, abortSignal: abortController.signal };
+
+    (async () => {
+      try {
+        const apiDefinition = integrationSettings.apiDefinition;
+        const celRequest: CelInputRequestBody = {
+          dataStreamName: integrationSettings.dataStreamName ?? '',
+          apiDefinition: apiDefinition ?? '',
```

Comment:
Isn;t `apiDefinition` mandatory?

Reply:
It is. I was referencing https://github.com/elastic/kibana/blob/main/x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/use_generation.tsx#L146-L148 which operated the same.

Reply:
Moved and renamed

Reply:
I have it optional there since if the FF isn't enabled it won't exist at all, but also, we want to allow the user to optionally do this flow, so it should always be optional. 

I made it required on the cel generation path, because it is needed to run there. But this object is for the integration, where it is currently and will be always optional to the user. 

Re log samples, I can see that being required, but I'd rather not change that in this PR, since it's already a lot of changes.

Reply:
Ok Then it makes sense..